### PR TITLE
cc-wrapper, bintools-wrapper: Remove now unneeded `set {+,-}u`

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

I noticed late in https://github.com/NixOS/nixpkgs/pull/72347 that I missed a few in the wrappers. This could cause more breakage since the `set +u` now "leaks" with out the stdenv cleanup, so I figured rather than last-minute add to the last PR, I'd make another.

https://hydra.nixos.org/jobset/nixpkgs/bash-no-undef-vars

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/tools/build-managers/cmake/setup-hook.sh [hunk 1]
```diff
@@ -143,7 +143,7 @@ makeCmakeFindLibs(){
           export CMAKE_LIBRARY_PATH="$CMAKE_LIBRARY_PATH${CMAKE_LIBRARY_PATH:+:}${flag:2}"
           ;;
         -F*)
-          export CMAKE_FRAMEWORK_PATH="$CMAKE_FRAMEWORK_PATH${CMAKE_FRAMEWORK_PATH:+:}${flag:2}"
+          export CMAKE_FRAMEWORK_PATH="${CMAKE_FRAMEWORK_PATH-}${CMAKE_FRAMEWORK_PATH:+:}${flag:2}"
```
[the remark this answers is not in the document]
Comment (Ericson2314) [reply pkgs/development/tools/build-managers/cmake/setup-hook.sh]: Oops! Good cathc

**Review by matthewbauer — APPROVED** [review]

File: pkgs/development/libraries/gdk-pixbuf/setup-hook.sh [hunk 2]
```diff
@@ -2,9 +2,9 @@ findGdkPixbufLoaders() {
 
 	# choose the longest loaders.cache
 	local loadersCache="$1/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache"
-	if [ -f "$loadersCache" ]; then
-		if [ -f "$GDK_PIXBUF_MODULE_FILE" ]; then
-			if [ $(cat "$loadersCache"|wc -l) -gt $(cat "$GDK_PIXBUF_MODULE_FILE"|wc -l) ]; then
+	if [[ -f "$loadersCache" ]]; then
+		if [[ -f "${GDK_PIXBUF_MODULE_FILE-}" ]]; then
+			if [[ "$(cat "$loadersCache" | wc -l)" < "$(cat "$GDK_PIXBUF_MODULE_FILE" | wc -l)" ]]; then
```
Comment (jtojnar) [inline pkgs/development/libraries/gdk-pixbuf/setup-hook.sh]: Should not this rather be `>`

File: pkgs/development/libraries/glib/setup-hook.sh [hunk 3]
```diff
@@ -1,9 +1,11 @@
 
 make_glib_find_gsettings_schemas() {
-    # For packages that need gschemas of other packages (e.g. empathy)
-    if [ -d "$1"/share/gsettings-schemas/*/glib-2.0/schemas ]; then
-        addToSearchPath GSETTINGS_SCHEMAS_PATH "$1/share/gsettings-schemas/"*
+  # For packages that need gschemas of other packages (e.g. empathy)
```
Comment (jtojnar) [inline pkgs/development/libraries/glib/setup-hook.sh]: Are not we supposed to use 4 space indent in shell scripts?

Comment (Ericson2314) [reply pkgs/development/libraries/glib/setup-hook.sh]: Are we?

Comment (jtojnar) [reply pkgs/development/libraries/glib/setup-hook.sh]: According to [this](https://nixos.org/nixpkgs/manual/#sec-syntax), yes.

Comment (Ericson2314) [reply pkgs/development/libraries/glib/setup-hook.sh]: Oh, thanks.

**Comment by Ericson2314** [comment]
OK, basically all the Darwin errors are due to unreated python and swift problems, and I just fixes the last linux errors I see, so I think this is ready for staging. I'll just watch staging if any darwin problems get uncovered.

**Comment by matthewbauer** [comment]
Yeah that sounds good

**Pull request merged**